src/swaps/magic_routing.rs

```diff
@@ -19,14 +22,64 @@ const LBTC_MAINNET_ASSET_HASH: &str =
     "6f0279e9ed041c3d710a9f57d0c02928416460c4b722ae3457a11eec381c526d";
 
 /// Decodes the provided invoice to find the magic routing hint.
-pub fn find_magic_routing_hint(invoice: &str) -> Result<Option<RouteHintHop>, Error> {
-    let invoice = Bolt11Invoice::from_str(invoice)?;
+pub fn find_magic_routing_hint(
+    invoice: &str,
+) -> Result<
+    Option<(
+        /* short_channel_id */ u64,
+        /* public_key */ String,
+    )>,
```

Comment:
Why get rid of the return type

Reply:
Because it's only applicable for BOLT11, this handles BOLT12 also